### ISSUE TITLE
Global seed variable

### ISSUE THREAD

**Issue by Gerhardsa0** [Safe-DS/Library#12]
### Is your feature request related to a problem? Please describe

For functions like ```Table.shuffle()``` who are not deterministic we do not have the ability to set a seed and get always the same but random output.

### Desired solution

We should implement a global variable which sets on all non-deterministic functions a seed, so we get always the same output for those functions.

Also all tests/functions should be updated and extended for non-deterministic functions.

### Additional context (optional)

so the user can use something like this:
```Python
# not deterministic
Table.shuffle()

# set var to to true after that all calls are deterministic

set_seed = 111

# now seed 111 will be used

# deterministic
Table.shuffle()
```

**Comment by lars-reimann** [comment]
Our functions should always be deterministic since we want the program execution to be reproducible. When we interact with `scikit-learn`, we should thus always specify the `random_seed` parameter of the functions we call (where applicable).

**Comment by guenterk** [comment]
1. Es geht darum, dass ALLES deterministisch ist, typischerweise, um zu testen.
2. Daher ist es am plausibelsten und konsistentesten, statt eines expliziten seeds einen Parameter des SAFE-DS-RUNNERs zu haben, der steuert ob deterministisch oder nicht. Typischerweise ein Boolean.

**Comment by lars-reimann** [comment]
Global state does not work well with memoization and hyperparameter optimization. We'll explicitly pass random seeds into the functions that require it, like `scikit-learn` and `Jax` are doing.
